tools/gorm/print_auto_migration.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"os"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/db"
+	"github.com/buildbuddy-io/buildbuddy/server/util/fileresolver"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"gorm.io/gorm"
+
+	bundle "github.com/buildbuddy-io/buildbuddy"
+)
+
+var (
+	dataSource = flag.String("data_source", "sqlite3:///tmp/buildbuddy.db", "The SQL database to connect to, specified as a connection string.")
+	outputPath = flag.String("output_path", "migration.txt", "The output file to write the gorm migration logs to.")
+)
+
+// This script prints the SQL commands that would be executed when running db.RunMigrations
+// It does not execute any of the commands
```

Comment:
Is it possible that some commands will be conditional on previous commands executing? I wonder if it would make sense to just let the commands execute.

If you are worried about modifying dev, you could maybe run a fresh mysql DB initialized to the current schema in dev (e.g. using Docker or maybe actions-setup-mysql to init a fresh DB, and running `SHOW TABLES` + `SHOW CREATE TABLE` statements against dev to recreate the current schema).

Or (simpler but less accurate) you could run the migration once at the current prod commit SHA against the empty mysql DB. 

Or (even simpler but more risky, maybe?) run against the actual DB but using a readonly connection, and just see if it fails due to trying to execute an ALTER TABLE statement but not being allowed to because of the readonly perms

Reply:
Modified to run the auto-migration on a db copy